sympy/solvers/solveset.py

```diff
@@ -2902,7 +2902,7 @@ def _solve_using_known_values(result, solver):
                             base = value_res.base_set
                             imgset_yes = (dummy_n, base)
                 # update eq with everything that is known so far
-                eq2 = eq.subs(res)
+                eq2 = eq.subs(res).expand()
```

Comment:
Would `_mexpand` be enough? or just `expand_mul`?

Reply:
There should be a test for this, too.

Reply:
Sir, can u please elaborate as to what should be the exact line.
Should it be eq2 = eq.subs(res).expand_mul() ?
Because i only know about the .expand() fuction.

Reply:
> Would `_mexpand` be enough? or just `expand_mul`?
 .expand() is better than using .expand_mul()

Reply:
No, those do not suffice to simplify the polynomial expression in this example.